nndownload/nndownload.py

```diff
@@ -2005,170 +2005,134 @@ def download_video_comments(
+                    "targets": [thread]
+                },
+                "additionals": {
+                    "res_from": -1000,
+                    "when": comments_from
                 }
-                
-                response = session.post(
-                    COMMENTS_THREAD_URL.format(api_server),
-                    json=payload,
-                    headers=headers
-                )
-                data = response.json()
-                
-                # Handle API errors, if any, with appropriate actions
-                error_code = data["meta"]["errorCode"] if "meta" in data and "errorCode" in data["meta"] else None
-                if error_code is not None:
-                    if error_code == "TOO_MANY_REQUESTS":
-                        output("Rate limited - waiting 60 seconds...\n", logging.INFO)
-                        time.sleep(COMMENTS_THREAD_COOLDOWN_S)
-                        continue
-                    elif error_code == "EXPIRED_TOKEN":
-                        output("Refreshing thread key...\n", logging.INFO)
-                        refresh_thread_key(session, video_id)
-                        continue
-                    elif error_code == "INVALID_TOKEN":
-                        raise AuthenticationException
-                    else:
-                        raise ParameterExtractionException(error_code)
-
-                # Update global comments count if higher
-                global_comment_count = data["data"]["globalComments"][0]["count"]
-                if global_comment_count and global_comment_count > COMMENTS_DATA_JSON["globalComments"][0]["count"]:
-                    COMMENTS_DATA_JSON["globalComments"][0]["count"] = global_comment_count
-                    COMMENTS_DATA_JSON["globalComments"][0]["id"] = thread["id"]
-
-                # Update thread comment count if first fetch
-                if fetched_comments_count == 0:
-                    thread_data["commentCount"] = data["data"]["threads"][0]["commentCount"]
-                    # If requesting all comments, we use the video's total comment count
-                    if _CMDL_OPTS.request_all_comments:
-                        total_comments = thread_data["commentCount"]
-                    else:
-                        # Otherwise, we use the limit provided or the video's total comment count, whichever is lower
-                        total_comments = min(thread_data["commentCount"], comments_limit)
-                    progress.update(task, total=total_comments)
-                    progress.update(task, description=f"Downloading {thread['fork']} comments")
-                    
-                thread_comments = data["data"]["threads"][0]["comments"]
-
-                if not thread_comments:
-                    # There are no comments before lastTime to fetch
-                    break
-                    
-                # If we got the same comments as last time, we should stop
```

Comment:
This seems really unnecessary, especially iterating over every comment. Isn't it really unlikely that would we retrieve the exact same request twice? If this is a concern, it should be sufficient to see if one specific last ID is present in both requests. I've taken this out for now